lib/spack/spack/extensions.py

```diff
@@ -11,10 +11,48 @@
 import types
 
 import llnl.util.lang
-import llnl.util.tty as tty
+import spack.cmd
```

Comment:
This seems like a circular import as `spack.cmd` also imports `spack.extensions`. I'd like to avoid them as they might cause trouble on module initialization.

Reply:
As primarily a compiled-language programmer that left me uneasy too, but I haven't seen any issues. I can tease the relevant items into a separate module to avoid the circularity, if you prefer.